### PR TITLE
[Host] - Adds API call to Confidence Switch action

### PR DESCRIPTION
**Pull request by drewjhart**
_Note: this is PR is one of a series of nested PRs that provide new features to `host`. They breakdown as such:
[[Host] - Base UI Updates in Advance of New Features #792](https://github.com/rightoneducation/righton-app/pull/792)
->[[Host] - Real Time Responses #793](https://github.com/rightoneducation/righton-app/pull/793)
->->[[Host] - Adds API call to Confidence Switch action #799](https://github.com/rightoneducation/righton-app/pull/799)
->->->[[Host] - Player Confidence Graph #800 ](https://github.com/rightoneducation/righton-app/pull/800)
->->->->[[Host] - Prettier #804](https://github.com/rightoneducation/righton-app/pull/804)_

**Issue:**

This PR provides updates to the `isConfidenceEnabled` value in the GameSession on the backend via the networking layer. Additionally, it cleans up the handling of `currentQuestionIndex` and simplifies the flow of the Start Game and Question Configuration phases of the pregame.

**Relevant Code:**
`handleConfidenceSwitchChange` in `GameSessionContainer.jsx`

### CODE REVIEW

File: central/yarn.lock [hunk 1]
```diff
@@ -3,9 +3,9 @@
```
Comment (drewjhart) [inline central/yarn.lock:4]: changes via: https://github.com/rightoneducation/righton-app/pull/797 or similar. already merged into dev

File: debug-panel/yarn.lock [hunk 2]
```diff
@@ -3,9 +3,9 @@
```
Comment (drewjhart) [inline debug-panel/yarn.lock:3]: changes via: https://github.com/rightoneducation/righton-app/pull/797 or similar. already merged into dev

File: host/yarn.lock [hunk 3]
```diff
@@ -3,9 +3,9 @@
```
Comment (drewjhart) [inline host/yarn.lock:4]: changes via: https://github.com/rightoneducation/righton-app/pull/797 or similar. already merged into dev

File: play/src/components/ScoreIndicator.tsx [hunk 4]
```diff
@@ -11,7 +11,30 @@ const ScorePill = styled('div')(({ theme }) => ({
   height: '22px',
   borderRadius: '23px',
   background: `${theme.palette.primary.highlightGradient}`,
-  zIndex: 1,
+  zIndex: 2,
+}));
+
```
Comment (drewjhart) [inline play/src/components/ScoreIndicator.tsx:16]: already merged with dev https://github.com/rightoneducation/righton-app/pull/788

File: play/yarn.lock [hunk 5]
```diff
@@ -8,9 +8,9 @@
   integrity sha512-1Yjs2SvM8TflER/OD3cOjhWWOZb58A2t7wpE2S9XfBYTiIl+XFhQG2bjy4Pu1I+EAlCNUzRDYDdFwFYUKvXcIA==
 
 "@adobe/css-tools@^4.0.1":
```
Comment (drewjhart) [inline play/yarn.lock:10]: already merged to dev via: https://github.com/rightoneducation/righton-app/pull/794

File: play/src/lib/HelperFunctions.tsx [hunk 6]
```diff
@@ -85,6 +86,41 @@ export const checkForSubmittedAnswerOnRejoin = (
...
+  // here, since we do not store time of last select in the backend (5 seconds would be negligible on refresh),
+  // we set the timeOfLastSelect to null to re-initialize the value instead of populating it with the previous value
+  const timeOfLastSelect = 0;
+  // adding dictionary to account for string casting for material UI components
+  if (
+    hasRejoined &&
+    (currentState === GameSessionState.CHOOSE_CORRECT_ANSWER ||
+      currentState === GameSessionState.CHOOSE_TRICKIEST_ANSWER)
+  ) {
+    if (!isNullOrUndefined(currentAnswer)) {
```
Comment (maniramezan) [inline play/src/lib/HelperFunctions.tsx:116]: nit: This can be part of the outer if statement as well

**Review by maniramezan — APPROVED** [review]

File: host/src/components/FooterStartGame.jsx [hunk 7]
```diff
@@ -11,7 +11,7 @@ const FooterStartGame = ({ handleStartGame, teamsLength, currentQuestionIndex })
       <Button
          disabled = {teamsLength <= 0 ? true: false} 
         className={classes.startGameButton} onClick={() => handleStartGame()}>
-        {currentQuestionIndex === null ? "Start Game" : "Next Question"}
+         Start Game
```
[the remark this answers is not in the document]
Comment (drewjhart) [reply host/src/components/FooterStartGame.jsx:14]: good catch, fixed

**Pull request merged**